Image sequencer, infragram, and RPi Cam for Raspberry Pi camera (wrong branch)

jywarren opened the pull request:
**Note:** this is opened against the `master` branch instead of the correct `main` branch. However another PR already exists (https://github.com/publiclab/pi-builder/pull/73) for this branch against `main` so we cannot repoint this PR. Keeping it open to preserve the discussion!

****

Thanks for opening a pull request! In this repository, opening a PR will initiate the generation of a new Raspberry Pi image, and create an image file you can download and use in your Raspberry Pi.

The changes you add to the pull request, such as adding software to install, will be run on the generated image.

For an example, see the software installed and configured in this pull request: https://github.com/publiclab/image-builder-rpi/pull/15/files

### Recipe

Use this space to describe what your "recipe" is intended to install and configure on a Raspberry Pi:




****

### Download instructions

Generating the image will take a few minutes. Once the image is prepared, and if it succeeded, you'll see a green checkmark at the bottom of the pull request. To download the image:

1. click the green checkmark; you'll go to a page at a URL like `https://gitlab.com/publiclab/image-builder-rpi/pipelines/########/builds`
2. On this page, click the `Jobs` tab, next to `Pipeline`
3. Click the green `Passed` button
4. Click `Download` in the right-hand sidebar
5. Unzip the `artifacts.zip` file, and also the `hypriotos-rpi-camera_web.img.zip` within it
6. Use a program like https://etcher.io/ to flash it to an SD card

You'll also be able to read the output of the image generation in this window.

We hope to create a bot to report back the completed image URL in each pull request. If you can help create such a bot, please contact us at:

https://github.com/publiclab/image-builder-rpi/issues/16

Thanks!

Note: a working image can be downloaded here:

https://gitlab.com/publiclab/pi-builder/-/jobs/121890022/artifacts/download

icarito committed:
Add rpicam software

icarito committed:
Add rpicam software

jywarren committed:
Install infragram

jywarren committed:
Install infragram

jywarren committed:
add image sequencer

jywarren committed:
add image sequencer

jywarren committed:
added buttons and new version of image sequencer

jywarren committed:
added buttons and new version of image sequencer

jywarren commented:
trying to do it from a branch in the same repo... based on #50

jywarren commented:
Download available!

https://gitlab.com/publiclab/pi-builder/-/jobs/118727487/artifacts/download

jywarren committed:
tweak to rebuild w/ new image sequencer

jywarren committed:
tweak to rebuild w/ new image sequencer

jywarren commented:
Rebuilding for new version of image sequencer!

jywarren commented:
New image available! https://gitlab.com/publiclab/pi-builder/-/jobs/121890022/artifacts/download

jywarren commented:
Hi @icarito i've just published a new `v0.2.0` version of https://github.com/publiclab/infragram on npm; if I want to rebuild with that version, should I just make a small arbitrary change here to trigger GitLab to generate a new image artifact? Thanks!

jywarren committed:
Trigger rebuild

jywarren committed:
Trigger rebuild

jywarren commented:
Same error as in #68 !

jywarren commented:
OK, this needs rebase now after #69 !

jywarren commented:
Rebased!

jywarren commented on builder/build.sh:
```diff
@@ -84,7 +84,7 @@ ls -alh /image_with_kernel_*.tar.gz
 # download the ready-made raw image for the RPi
 if [ ! -f "${BUILD_RESULT_PATH}/${RAW_IMAGE}.zip" ]; then
-  wget -q -O "${BUILD_RESULT_PATH}/${RAW_IMAGE}.zip" "https://github.com/hypriot/image-builder-raw/releases/download/${RAW_IMAGE_VERSION}/${RAW_IMAGE}.zip"
+  wget -q -O "${BUILD_RESULT_PATH}/${RAW_IMAGE}.zip" "https://jenkins.laboratoriopublico.org/job/image-builder-raw/ws/${RAW_IMAGE}.zip"
```
@icarito can you check if this should be here? It exists on https://github.com/publiclab/pi-builder/pull/40/files but not sure if it's an artifact of rebasing.

jywarren commented on builder/chroot-script.sh:
```diff
@@ -94,12 +94,9 @@ export DEST
-get_gpg "${PACKAGECLOUD_FPR}" "${PACKAGECLOUD_KEY_URL}"
-
-echo 'deb https://packagecloud.io/Hypriot/rpi/debian/ stretch main' > /etc/apt/sources.list.d/hypriot.list
+# set up hypriot rpi repository for raspbian specific packages
```
Same here... is this right for this PR or is this reverting something?

jywarren commented on builder/files/etc/rc.local:
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh -e
```
And one more spot I'm uncertain - this doesn't exist in https://github.com/publiclab/pi-builder/pull/40/files for example, which builds properly and is supposed to be almost the same as this PR.

jywarren commented on versions.config:
```diff
@@ -4,14 +4,15 @@ ROOTFS_TAR_CHECKSUM="d1e7e6d48a25b4a206c5df99ecb8815388ec6945e4f97e78413d5a80778
 RAW_IMAGE="rpi-raw.img"
-RAW_IMAGE_VERSION="v0.2.2"
-RAW_IMAGE_CHECKSUM="2fbeb13b7b0f2308dbd0d82780b54c33003ad43d145ff08498b25fb8bbe1c2c6"
+RAW_IMAGE_VERSION="master"
```
Ah, one more, apologies. this is the last spot.

jywarren reviewed (commented):
@icarito would you mind checking if my rebase introduced a bunch of unnecessary stuff vs. https://github.com/publiclab/pi-builder/pull/40/files ? Thank you!

icarito commented:
Sorry I just got around to check this! 

I think that the issue is you're rebasing into `master` branch, but we're using `main` branch.

jywarren commented:
Ah, this makes sense, but actually all our PRs are against `master` - maybe we did the transition to `main` after these PRs were open? Should we merge to `master` from `main` to sync them, or are they forked in two different directions?

icarito commented:
You're right our PR's are for `master` branch anyway. I'll review with more detail.

icarito committed:
Attempt rebase

Merge branch 'image-sequencer' of https://gitlab.com/publiclab/pi-builder into image-sequencer

icarito commented:
Hey, i figured out the issue is that you force-pushed and this stalled the repository synchronization. It's building now, fingers crossed!

icarito committed:
Fix build!

icarito commented:
I fixed the kernel reference from upstream, now running into:
```
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! gl@4.2.2 install: `prebuild-install || node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the gl@4.2.2 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above
```
Perhaps this was faced by ImageSequencer before?

jywarren commented:
Ah ah ah! That's good news. It's a gl component, so using the graphics
card. Let me ask @Divy123 about compiling gl on an arm7 processor... On a
Raspberry Pi.  Can you link us to the exact error output? Thanks!!

On Tue, Jul 16, 2019, 4:32 AM Sebastian Silva <notifications@github.com>
wrote:

> I fixed the kernel reference from upstream, now running into:
>
> npm ERR! code ELIFECYCLE
> npm ERR! errno 1
> npm ERR! gl@4.2.2 install: `prebuild-install || node-gyp rebuild`
> npm ERR! Exit status 1
> npm ERR!
> npm ERR! Failed at the gl@4.2.2 install script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above
>
> Perhaps this was faced by ImageSequencer before?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/pi-builder/pull/53?email_source=notifications&email_token=AAAF6JY3UPQFY6WJNVP7FBLP7WBODA5CNFSM4GC54ZDKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2ADWUA#issuecomment-511720272>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAF6J3ZKYMMQBO3JZ7KXMLP7WBODANCNFSM4GC54ZDA>
> .
>

jywarren committed:
sudo apt-get install -y build-essential libxi-dev libglu1-mesa-dev libglew-dev pkg-config

jywarren commented on builder/chroot-script.sh:
```diff
@@ -205,10 +208,44 @@ lighttpd-enable-mod fastcgi-php
+
+# install infragram in the web public folder:
+# prerequisites:
+sudo apt-get install -y build-essential libxi-dev libglu1-mesa-dev libglew-dev pkg-config
```
Ok @icarito do I need to rebase this again?

jywarren replied:
Ah looks like no, it's building now!

Divy123 commented:
@jywarren is there something I can help here?

jywarren commented:
@divy123 is python a requirement to build headless-gl?

```
gl@4.2.2 install /var/www/image-sequencer/node_modules/gl
> prebuild-install || node-gyp rebuild

prebuild-install WARN install No prebuilt binaries found (target=9.7.1 runtime=node arch=arm libc= platform=linux)
gyp ERR! configure error 
gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.
gyp ERR! stack     at PythonFinder.failNoPython (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:483:19)
gyp ERR! stack     at PythonFinder.<anonymous> 
```

Divy123 commented:
yes, we do require!!

jywarren committed:
Add python2.7

icarito committed:
Fix typo

jywarren commented:
Hmm:

```

> gl@4.2.2 install /var/www/image-sequencer/node_modules/gl
> prebuild-install || node-gyp rebuild

prebuild-install WARN install No prebuilt binaries found (target=9.7.1 runtime=node arch=arm libc= platform=linux)
gyp ERR! configure error 
gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.
gyp ERR! stack     at PythonFinder.failNoPython (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:483:19)
gyp ERR! stack     at PythonFinder.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:397:16)
gyp ERR! stack     at F (/usr/local/lib/node_modules/npm/node_modules/which/which.js:68:16)
gyp ERR! stack     at E (/usr/local/lib/node_modules/npm/node_modules/which/which.js:80:29)
gyp ERR! stack     at /usr/local/lib/node_modules/npm/node_modules/which/which.js:89:16
gyp ERR! stack     at /usr/local/lib/node_modules/npm/node_modules/which/node_modules/isexe/index.js:42:5
gyp ERR! stack     at /usr/local/lib/node_modules/npm/node_modules/which/node_modules/isexe/mode.js:8:5
gyp ERR! stack     at FSReqWrap.oncomplete (fs.js:170:21)
gyp ERR! System Linux 4.19.23-coreos-r1
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /var/www/image-sequencer/node_modules/gl
gyp ERR! node -v v9.7.1
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok 
npm WARN The package eslint is included as both a dev and production dependency.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"arm"})
```

jywarren commented on builder/chroot-script.sh:
```diff
@@ -205,10 +208,43 @@ lighttpd-enable-mod fastcgi-php
+
+# install infragram in the web public folder:
+# prerequisites:
+sudo apt-get install -y build-essential libxi-dev libglu1-mesa-dev libglew-dev pkg-config python2.7 python2.7-dev
```
Maybe this doesn't install as `python`?

Divy123 replied:
> Maybe this doesn't install as python?

I think this may be needed:
sudo apt-get install -y build-essential libxi-dev libgl1-mesa-dev libglew-dev pkg-config python2.7 python2.7-dev. Just confirming!!

icarito committed:
Try regular python package

jywarren committed:
changed to python2.7 python2.7-dev

jywarren commented:
Hmm, switching back to python 2.7, but really just trying to not have a force push so that this will build properly...

jywarren commented:
We may need to close this and move to a new PR which is against `main`... ok, opened #73 against `main` branch!

jywarren committed:
install spectral-workbench.js

jywarren committed:
update menu

jywarren committed:
back to python not python2.7

jywarren committed:
revert builder/build.sh

jywarren committed:
revert versions.config

icarito commented:
Hey, I just went to check the Gitlab repository configuration and I found I could modify the CI timeout - I set it to 4 hours! So maybe let's try again with the build that timed out? Thanks and sorry I didn't know that was possible!

icarito commented:
Here I restarted the build from last week:
https://gitlab.com/publiclab/pi-builder/-/jobs/264078267

icarito commented:
https://gitlab.com/publiclab/pi-builder/-/jobs/264059463 (revert builder/build.sh) timed out at one hour because it started before I changed the setting (the max is 3 hours). I'm restarting this build!

fennsen64 commented:
Testimage

jtravs commented:
Hi, can someone tell me where I should download this image? There is no green check mark?

jywarren commented:
Hi! It looks like the build didn't correctly complete -- so it shows a red X instead. The link is under the word `Details` below -- here it is: https://gitlab.com/publiclab/pi-builder/-/pipelines/74183616

The error was: 

`ln: failed to create symbolic link '/sys/fs/cgroup/systemd/name=systemd': Operation not permitted`

In the log here: https://gitlab.com/publiclab/pi-builder/-/jobs/264061251

Not sure why that happened! 

It looks like even in the new issue we moved to from here, we are seeing some issues as well: https://github.com/publiclab/pi-builder/pull/73
